### PR TITLE
Warn when http sources are used in PackageReference restore operations

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1358

Regression? Last working version:

## Description

Pretty straightforward. Contains the same warning as https://github.com/NuGet/NuGet.Client/pull/4552, but obviously with a code since it's restore.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [x] Documentation PR or issue filled https://github.com/NuGet/docs.microsoft.com-nuget/issues/2717
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Review by kartheekp-ms — DISMISSED** [review]
Maybe update the PR title saying this fix is specifically for `PackageReference` style projects.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs [hunk 1]
```diff
@@ -214,6 +214,18 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                     _success = false;
                 }
 
+                if (_request.Project?.RestoreMetadata != null)
+                {
+                    foreach (var source in _request.Project.RestoreMetadata.Sources)
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:219]: does this collection exclude disabled sources?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs:219]: nope, only enabled.

**Comment by nkolev92** [comment]
Fixed the title, nice catch.

**Review by jeffkl — DISMISSED** [review]

**Review by kartheekp-ms — APPROVED** [review]

**Comment by davidcorrigan714** [comment]
@nkolev92 We just hit this today, is there any official announcement or timeline about the HTTP deprecation? We've got an old server we'll be migrating off of in the near future and some concrete dates would help us with planning.

**Comment by joeltankam** [comment]
Same here, could you please give the planned Visual Studio / NuGet release for which non-HTTPS access will be completely removed ?

**Comment by japj** [comment]
Please note also that there does not seem to be any help for NU1803 on the Microsoft docs pages (e.g. if you click on the NU1803 error from VisualStudio it results in an "Oops! No F1 help match was found" webpage)

**Pull request merged**